lib/shared/bucketing-assembly-script/assembly/types/dvcUser.ts

```diff
@@ -43,10 +43,16 @@ export class DVCUser extends JSON.Obj implements DVCUserInterface {
         this.appBuild = getF64FromJSONOptional(user, 'appBuild', -1)
 
         const customData = user.getObj('customData')
-        this.customData = (customData && customData.isObj) ? customData : null
+        if (!isFlatJSONObj(customData)) {
+            throw new Error(`DVCUser customData not a flat JSON Object`)
```

Comment:
should we be more descriptive with this error? `customData can't contain nested objects or arrays`?